panoptes_aggregation/panoptes/userify.py

```diff
@@ -0,0 +1,99 @@
+from collections import Iterable
+from json import dumps as jsonify
+from os import environ
+from flask import request
+from panoptes_client import Panoptes, User
+import requests
+from sys import modules
+
+class ConfigurationError(Exception):
+    pass
+
+this = modules[__name__]
+
+this.known_params = [
+    'destination'
+]
+
+this.connected = False
+
+this.users = {}
+
+this.destinations = {
+    'mast': 'https://mast-forward.zooniverse.org/',
+    'mockable': 'https://demo1580318.mockable.io/mast'
+}
+
+def userify():
+    if request.method == 'GET':
+        return 'Usage'
+    elif request.method in ['POST', 'PUT']:
+        find_fields = _discover_fields(request.args)
+        target_object = request.get_json()
+        destination = request.args.get('destination')
+
+        _stuff_object(target_object, find_fields)
+
+        if destination:
+            _forward_contents(target_object, destination)
+
+        return jsonify(target_object)
+
+def _forward_contents(payload, destination):
+    if destination not in this.destinations:
+        raise ConfigurationError('Unknown destination')
+
+    requests.post(url=this.destinations[destination], json=payload)
+    return
+
+def _stuff_object(target_object, find_fields):
+    for key in target_object.keys():
+        if type(target_object[key]) is dict:
+            _stuff_object(target_object[key], find_fields)
+
+    user_ids = _discover_user_ids(target_object)
+    for user_id in user_ids:
+        user = _retrieve_user(user_id)
+        target_object['users'] = target_object.get('users', [])
+        target_object['users'].append(_build_user_hash(user,find_fields))
+
+def _discover_fields(request_args):
+    raw_list = request_args.to_dict().keys()
+    return [ i for i in raw_list if i not in this.known_params ]
+
+def _discover_user_ids(target_object):
+    user_ids = []
+    if 'user_ids' in target_object:
+        user_ids = target_object['user_ids']
+    if 'user_id' in target_object:
+        user_ids.append(target_object['user_id'])
+
+    return _unique(_flatten(user_ids))
+
+def _build_user_hash(user, find_fields):
+    user_hash = { 'id': user.id }
+    for key in find_fields:
+        user_hash[key] = user.__getattr__(key)
+    return user_hash
+
+def _unique(l):
+    return list(set(l))
+
+def _flatten(l):
+    for el in l:
+        if isinstance(el, Iterable) and not isinstance(el, (str, bytes)):
+            yield from _flatten(el)
+        else:
+            yield el
+
+def _retrieve_user(user_id):
+    if user_id in this.users:
+        user = this.users[user_id]
+    else:
+        if this.connected == False:
```

Comment:
Comparisons to `True` and `False` are redundant:

```python
if not this.connected:
```

Reply:
oh that is better! done